### PR TITLE
Containers

### PR DESCRIPTION
**Pull request by taylor-vann**
Rust libraries to generate postgres and redis containers based on templates.

### CODE REVIEW

**Comment by taylor-vann** [comment]
resolves: #1

**Comment by andrewblum** [comment]
Seems neat :) Help me understand the benefits over just making and using container spec and container compose files directly?

**Comment by taylor-vann** [comment]
:D mainly a couple benefits:
- consistent containers across org, rather than editing a multiple container files or repositories at once
- support multiple containers, there could be multiple containers for a dev / prod or emulated distributed systems
- separates responsibility so later a `setup.sh` or `onboard.sh` can run `generate_postgres_container` or `generate_redis_container` modules where needed

This repo is a "script" to create container files that will be committed to multiple repositories.

Like if two repositories `a` and `b` need a redis container for testing or production or whatever. This repository would generate identical redis container files for both. Which reduces differences in container implementations as a debug issue.

File: postgres_container/generate_postgres_templates/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    let args = match env::args().nth(1) {
+        Some(a) => path::PathBuf::from(a),
+        None => return println!("argument error:\nconfig file not found."),
+    };
+    let config = match config::from_filepath(&args).await {
+        Ok(c) => c,
+        Err(e) => return println!("configuration error:\n{}", e),
+    };
+
+    if let Err(e) = generate_data_dir(&config).await {
```
Comment (taylor-vann) [inline postgres_container/generate_postgres_templates/src/main.rs:19]: looking this over, the "functional" part of this `main` should be it's own rust lib. That way other modules could use this.

**Comment by taylor-vann** [comment]
I gotta admit the simplicity of one set of container files is nice. And I've been stuck in distributed land for a while. And maintaining stuff is a chore. Let's do a single bunch of template files :D

I think I can set all the environment files in a single `podman-compose` file. Upcoming changes.

**Comment by taylor-vann** [comment]
@andrewblum re-requesting a review

changes:
- removed all the template generation
- use env vars to run `podman-compose`

This ties `redis` and `postgres` into the same "system" but for now I think that's okay :)

**Pull request merged**